.github/workflows/update-test-weights.yml

```diff
@@ -0,0 +1,145 @@
+name: Update Test Weights
+
+on:
+  schedule:
+    # Run every Sunday at 2am UTC
+    - cron: "0 2 * * 0"
+
+  workflow_dispatch:
+    inputs:
+      run_count:
+        description: "Number of recent CI runs to analyze (default: 3)"
+        required: false
+        default: "3"
+        type: string
+
+permissions:
+  contents: write # To create branches and commits
+  actions: read # To fetch workflow runs and artifacts
+  pull-requests: write # To create pull requests
+
+jobs:
+  update-weights:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    steps:
+      - name: Checkout repository
```

Comment:
Do we wish to add a check for repository name so this does not run for all forks? Like we don't want https://github.com/acryldata/datahub to have PRs created